query/compile.go

```diff
@@ -115,6 +115,17 @@ func Compile(stmt *influxql.SelectStatement, opt CompileOptions) (Statement, err
 	c.stmt.TimeAlias = c.TimeFieldName
 	c.stmt.Condition = c.Condition
 
+	defer func() {
+		if e := recover(); e != nil && err == nil {
```

Comment:
I did a quick search but didn't see where the panic originates from. Is there Influx code panicing somewhere that we should change to return an error?

Reply:
The code that panics is type assertions pretty deep in influxql - I would have to change a bunch of code to switch to errors instead of panics.